sdk/metric/aggregation/aggregation.go

```diff
@@ -32,9 +32,13 @@ type Aggregation interface {
 
 	// Copy returns a deep copy of the Aggregation.
 	Copy() Aggregation
+	// DO NOT CHANGE: any modification will not be backwards compatible and
```

Comment:
We do not allow user-defined aggregations. Therefore should we add a following notice?

> Warning: methods may be added to this interface in minor releases.

Reply:
This is an exported interface with exported methods. Modifying the exported methods in any way is a breaking change once they are released stability regardless of if we allow user defined aggregations or not. This comment is the conclusion of the community decision in #1714 to ensure we make it clear to future developers to not introduce breaking changes here.

Reply:
My comment was NOT about this line in particular but about adding a comment referenced below

https://github.com/open-telemetry/opentelemetry-go/blob/2899fcfdcaf0d5499d2659b93cee7e609da8c65d/CONTRIBUTING.md#L473-L476

As far as I understand in this case we could add a method to this interface in future as I think it falls into "Internal Only Interface Implementations".

Reply:
As far as I remember we should not allow user defined aggreagations. Right now, I think it is possible to hack (type embedding + different implementation of exported methods) to provide custom ones.

Maybe we should move the aggregation to `sdk/metrics` package and unexport all methods to not make it possible (similar approach to option interface).

We could also consider adding "type constraints" to only allow types defined by the SDK.

Reply:
Resolved: https://github.com/open-telemetry/opentelemetry-go/issues/4432